src/cmd/tools/doctest/doctest.go

```diff
@@ -0,0 +1,171 @@
+	}
+
+	validationRegEx, operationRegEx, err := bashScript(f, file)
+	if err != nil {
+		log.Fatalf("unable to create bash script: %v", err)
+	}
+
+	if err := markdownFile(f, file, validationRegEx, operationRegEx); err != nil {
+		log.Fatalf("unable to create markdown file: %v", err)
+	}
+}
+
+func markdownFile(contents []byte, fileName string, validationRegEx *regexp.Regexp, opRegex *regexp.Regexp) error {
+	// create new markdown file based on name of source file
+	out, err := os.Create(strings.Replace(fileName, ".source", "", -1))
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	out.Chmod(0644)
+
+	// find all of the operations in the source file
+	matches := opRegex.FindAll(contents, 100)
+	for _, match := range matches {
+		validations := validationRegEx.FindAll(match, 100)
+		if len(validations) == 0 {
+			continue
+		}
+
+		// remove the validations and if the operation block is empty, replace it with nothing
+		noValidationBlock := bytes.Replace(match, validations[0], nil, -1)
+
+		if string(noValidationBlock) == fmt.Sprintf(operationOpen+"\n"+operationClose) {
+			contents = bytes.Replace(contents, match, []byte(""), -1)
+		}
+	}
+
+	// convert tags to markdown syntax
+	OpOpen := regexp.MustCompile(operationOpen)
+	OpClose := regexp.MustCompile(operationClose)
+	OpJSONOpen := regexp.MustCompile(operationJSONOpen)
+	OpJSONClose := regexp.MustCompile(operationJSONClose)
+
+	contents = OpOpen.ReplaceAll(contents, []byte("```"))
+	contents = OpClose.ReplaceAll(contents, []byte("```"))
+	contents = OpJSONOpen.ReplaceAll(contents, []byte("```json"))
+	contents = OpJSONClose.ReplaceAll(contents, []byte("```"))
+	contents = validationRegEx.ReplaceAll(contents, nil)
+	out.Write(contents)
+
+	return nil
+}
+
+func bashScript(contents []byte, fileName string) (*regexp.Regexp, *regexp.Regexp, error) {
+	// create bash script file based on name of source file
+	scriptFileName := fmt.Sprintf("%s%s", strings.Replace(fileName, ".md.source", "", -1), ".sh")
+	script, err := os.Create(scriptFileName)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer script.Close()
+
+	script.Chmod(0644)
+
+	// todo(braskin): figure out a way to make the script template dynamic
+	scriptTemplate, err := ioutil.ReadFile("script_template.txt")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	script.Write(scriptTemplate)
+
+	operationRegEx := regexp.MustCompile(operationRegEx)
+	validationRegEx := regexp.MustCompile(validationRegEx)
+
+	// find all of the operations in the source file
+	matches := operationRegEx.FindAll(contents, 100)
+
+	for _, match := range matches {
+		// find any validations
+		validationRaw := validationRegEx.Find(match)
+
+		// write operations to script
+		match = validationRegEx.ReplaceAll(match, nil)
+		operation := removeTags(string(match), []string{operationOpen, operationClose, operationJSONOpen, operationJSONClose})
+		script.WriteString(operation)
+
+		// write validations to script
+		if len(validationRaw) > 0 {
```

Comment:
Do we want to error out if there is more than one validation block per operation block?

Reply:
your call :)